views/pages/ConfigurationsPage/index.jsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from 'react'
+import { Box, Button } from '@mui/material'
+import { Add } from '@mui/icons-material'
+import { useOutletContext, Link } from 'react-router-dom'
+
+import ConfigurationsTable from '../../tables/ConfigurationsTable'
+import PageHeader from '../../components/PageHeader'
+import { UsersModel } from '../../models'
+import api from '../../api'
+import ShareConfigurationForm from '../../forms/ShareConfigurationForm'
+import { routes } from '../../routes/routes'
+
+const ConfigurationsPage = () => {
+  const {
+    configurations,
+    navigate,
+    setNotification,
+    setConfigurations,
+    setUser,
+    user,
+    users,
+  } = useOutletContext()
+  const { uid, preferences } = user
+  const [open, setOpen] = useState(false)
+  const [options, setOptions] = useState([])
+  const [formValues, setFormValues] = useState([])
+  const [currentConfigId, setCurrentConfigId] = useState('')
+
+  const handleNotification = (message) =>
+    setNotification({ open: true, message })
+  const onSubmit = async (data) => {
+    try {
+      const configAttributes = {
+        readers: UsersModel.createReadersList(data.readers),
+      }
+      await api.userConfigurations.update(
+        uid,
+        currentConfigId,
+        configAttributes
+      )
+
+      await loadAllConfigurations(uid)
+
+      closeForm()
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const loadAllConfigurations = async (userId) => {
+    try {
+      const configurations = await api.userConfigurations.all(userId)
+
+      setConfigurations(configurations)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const closeForm = () => {
+    setCurrentConfigId('')
+    setOpen(false)
+    setOptions([])
+    reset()
```

Comment:
where is reset coming from?

Reply:
good find
[3f060e8](https://github.com/AMP-SCZ/dpdash/pull/487/commits/3f060e8279d72eb848b54d8b7eb594892d7449a8)